shotover-proxy/src/message/mod.rs

```diff
@@ -87,14 +87,14 @@ pub struct RawMessage {
 
 #[derive(PartialEq, Debug, Clone)]
 pub enum ASTHolder {
-    SQL(Statement),
+    SQL(Box<Statement>),
     Commands(Value), // A flexible representation of a structured query that will naturally convert into the required type via into/from traits
 }
 
 impl ASTHolder {
     pub fn get_command(&self) -> String {
         match self {
-            ASTHolder::SQL(statement) => match statement {
+            ASTHolder::SQL(statement) => match &**statement {
```

Comment:
Is the borrow needed here?

Reply:
yeah, the borrow is needed.
self is borrowed so we need to also borrow statement

Reply:
wait, I tested it and it wasnt needed!

Reply:
It seems the reference only needs to readded for `&mut`, not sure why :shrug: